### PR TITLE
[Merged by Bors] - feat: port Analysis.Analytic.Basic

### PR DESCRIPTION
**Pull request by Ruben-VandeVelde**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by int-y1** [comment]
todo:

* [ ] (optional) in the docs, find the translation for `formal_multilinear_series.mk_pi_field`
* [ ] `nnnorm_changeOriginSeries_le_tsum` sorry

File: Mathlib/Analysis/Analytic/Basic.lean [hunk 1]
```diff
@@ -373,26 +381,29 @@ variable {f g : E → F} {p pf pg : FormalMultilinearSeries 𝕜 E F} {x : E} {r
 /-- Given a function `f : E → F` and a formal multilinear series `p`, we say that `f` has `p` as
 a power series on the ball of radius `r > 0` around `x` if `f (x + y) = ∑' pₙ yⁿ` for all `‖y‖ < r`.
 -/
-structure HasFpowerSeriesOnBall (f : E → F) (p : FormalMultilinearSeries 𝕜 E F) (x : E) (r : ℝ≥0∞) :
-  Prop where
+structure HasFPowerSeriesOnBall (f : E → F) (p : FormalMultilinearSeries 𝕜 E F) (x : E) (r : ℝ≥0∞) :
+    Prop where
+  /-- The provided radius is at mots the `FormalMultilinearSeries.radius`. -/
```
Comment (jcommelin) [inline Mathlib/Analysis/Analytic/Basic.lean:386]: This docstring contained a typo. Adding docs is not needed during the port, and should be avoided for reasons like these...
```suggestion
```

File: Mathlib/Analysis/Analytic/Basic.lean [hunk 2]
```diff
@@ -373,26 +381,29 @@ variable {f g : E → F} {p pf pg : FormalMultilinearSeries 𝕜 E F} {x : E} {r
...
   r_le : r ≤ p.radius
+  /-- The provided radius is positive. -/
```
Comment (jcommelin) [inline Mathlib/Analysis/Analytic/Basic.lean:388]:
```suggestion
```

File: Mathlib/Analysis/Analytic/Basic.lean [hunk 3]
```diff
@@ -373,26 +381,29 @@ variable {f g : E → F} {p pf pg : FormalMultilinearSeries 𝕜 E F} {x : E} {r
...
   r_pos : 0 < r
-  HasSum :
-    ∀ {y}, y ∈ EMetric.ball (0 : E) r → HasSum (fun n : ℕ => p n fun i : Fin n => y) (f (x + y))
-#align has_fpower_series_on_ball HasFpowerSeriesOnBall
+  /-- For every `y` in the ball of radius `r` centered at `0`, `f (x + y) = ∑' pₙ yⁿ`. -/
```
Comment (jcommelin) [inline Mathlib/Analysis/Analytic/Basic.lean:390]:
```suggestion
```

File: Mathlib/Analysis/Analytic/Basic.lean [hunk 4]
```diff
@@ -731,177 +736,180 @@ theorem HasFpowerSeriesOnBall.isBigO_image_sub_image_sub_deriv_principal
...
+          -- porting note: `pi_norm_const` was `pi_norm_const (_ : E)`
+          simpa only [Fintype.card_fin, pi_norm_const, Prod.norm_def, Pi.sub_def,
             Prod.fst_sub, Prod.snd_sub, sub_sub_sub_cancel_right] using
             (p <| n + 2).norm_image_sub_le (fun _ => y.1 - x) fun _ => y.2 - x
         _ = ‖p (n + 2)‖ * ‖y - (x, x)‖ ^ n * (↑(n + 2) * ‖y - (x, x)‖ * ‖y.1 - y.2‖) := by
           rw [pow_succ ‖y - (x, x)‖]
           ring
-        _ ≤ C * a ^ (n + 2) / r' ^ (n + 2) * r' ^ n * (↑(n + 2) * ‖y - (x, x)‖ * ‖y.1 - y.2‖) := by
-          apply_rules [mul_le_mul_of_nonneg_right, mul_le_mul, hp, pow_le_pow_of_le_left, hy'.le,
+        -- porting note: the two `↑` in `↑r'` are new, without them, Lean errors with `HDiv, HMul`
```
Comment (jcommelin) [inline Mathlib/Analysis/Analytic/Basic.lean:756]: I don't understand what "errors with `HDiv, HMul`" means? Is that the literal error message?

Comment (adomani) [reply Mathlib/Analysis/Analytic/Basic.lean]: Johan, I think that this is a porting note that I added: now it says, more explicitly
```lean
porting note: the two `↑` in `↑r'` are new, without them, Lean fails to synthesize
instances `HDiv ℝ ℝ≥0 ?m` or `HMul ℝ ℝ≥0 ?m`
```
Is this clearer now?

Comment (jcommelin) [reply Mathlib/Analysis/Analytic/Basic.lean]: Yes, thanks a lot

**Pull request closed without merging**